File: Example_Data/jupyter_exercise_materials/analyze_health_and_income.R
######################
#
# Import World Development Indicators
# and look at the relationship between income
# and health outcomes across countries
# 
######################

# Download World Development Indicators
wdi = read.csv("https://media.githubusercontent.com/media/nickeubank/MIDS_Data/master/World_Development_Indicators/wdi_small_tidy_2015.csv")

# Get Mortality and GDP per capita for 2015
wdi$loggdppercap = log(wdi[['GDP.per.capita..constant.2010.US..']])

# Plot
library(ggplot2)
ggplot(wdi, 
       aes(x=loggdppercap, 
           y=Mortality.rate..under.5..per.1.000.live.births.)
       ) + geom_point() + 
       geom_label(aes(label=Country.Name)) + 
       geom_smooth()

File: Example_Data/jupyter_exercise_materials/analyze_health_and_income.py
######################
#
# Import World Development Indicators
# and look at the relationship between income
# and health outcomes across countries
#
######################

import pandas as pd
import numpy as np

# Download World Development Indicators
wdi = pd.read_csv(
    "https://media.githubusercontent.com/"
    "media/nickeubank/MIDS_Data/"
    "master/World_Development_Indicators/wdi_small_tidy_2015.csv"
)

# GDP Per Capita has a REALLY long right tail, so we want to log it for readability.
wdi["Log GDP Per Capita"] = np.log(wdi["GDP per capita (constant 2010 US$)"])

# Plot
import seaborn.objects as so
import seaborn as sns
from matplotlib import style

my_chart = (
    so.Plot(
        wdi, x="Log GDP Per Capita", y="Mortality rate, under-5 (per 1,000 live births)"
    )
    .add(so.Line(), so.PolyFit(order=2))
    .add(so.Dot())
    .label(title="Log GDP and Under-5 Mortality")
    .theme({**style.library["seaborn-whitegrid"]})
)

my_chart

print("Done!")
